cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java

```diff
@@ -1834,7 +1834,7 @@ private String generatePerBrokerConfiguration(NodeRef node, KafkaPool pool, Map<
                 .withStrimziMetricsReporter(strimziReporterMetrics)
                 .withTieredStorage(cluster, tieredStorage)
                 .withQuotas(cluster, quotas)
-                .withUserConfiguration(configuration, node.broker() && ccMetricsReporter != null)
+                .withUserConfiguration(configuration, node.broker() && ccMetricsReporter != null, node.broker() && strimziReporterMetrics != null)
```

Comment:
Remember that `strimziReporterMetrics` can be not null but disabled due to validation errors. This should be:

```sh
.withUserConfiguration(
    configuration,
    node.broker() && ccMetricsReporter != null,
    strimziReporterMetrics != null && strimziReporterMetrics.isEnabled()
)
```

Reply:
Refactored